include/RAJA/pattern/reduce.hpp

```diff
@@ -64,6 +65,67 @@
 namespace RAJA
 {
 
+namespace reduce
```

Comment:
I'm not against this change, but I would like to quantify the performance impact of having these operator structs.